indico/modules/events/surveys/controllers/display.py

```diff
@@ -50,7 +53,7 @@ def view_class(self):
 class RHSurveyList(RHSurveyBaseDisplay):
     def _process(self):
         surveys = (Survey.query.with_parent(self.event_new)
-                   .filter(Survey.is_visible)
+                   .filter(Survey.is_visible, ~Survey.private)
```

Comment:
I agree with @ThiefMaster on this. If you've started it already, then it's useful that you can continue it without having to use the link again.